src/main/java/com/alliander/equigy/client/api/ActivatedEnergyClient.java

```diff
@@ -0,0 +1,86 @@
+
+    final private URI baseUri;
+
+    public ActivatedEnergyClient(URI baseUri) {
+        this.baseUri = baseUri;
+    }
+
+    public List<ActivatedEnergy> getByQuery(EAN18 accountingPointId, LocalDate queryDate, String bearerToken) {
+        try {
+            final HttpURLConnection connection = (HttpURLConnection) getByQueryUrl(accountingPointId, queryDate).openConnection();
+            connection.setRequestProperty("Authorization", "Bearer " + bearerToken);
+            connection.setRequestProperty("Accept", "application/json");
+
+            if (connection.getResponseCode() != 200) {
+                throw new EquigyException("Unexpected HTTP status code: " + connection.getResponseCode());
+            } else {
+                final String contentType = connection.getHeaderField("Content-Type");
+                if (contentType == null || !contentType.startsWith("application/json") || !contentType.contains("charset=utf-8")) {
+                    throw new EquigyException("Unexpected Content-Type in response: " + contentType);
+                }
+            }
+
+            final StringBuilder data = new StringBuilder();
+            try (final InputStreamReader reader = new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8)) {
+                final char[] buf = new char[1024];
+                for (int n = reader.read(buf); n > -1; n = reader.read(buf)) {
+                    data.append(buf, 0, n);
+                }
+            }
+
+            return Json.read(data.toString()).asJsonList().stream()
+                    .map(ActivatedEnergy::new)
+                    .collect(Collectors.toList());
+        } catch (IOException | NullPointerException | Json.MalformedJsonException e) {
+            throw new EquigyException(e);
+        }
+    }
+
+    public List<ActivatedEnergy> getByQuery(EAN18 accountingPointId, LocalDate queryDate, OAuthToken token) {
+        if (!token.getTokenType().equals("Bearer")) {
+            throw new EquigyException("OAuth token of unknown token type");
+        }
+        if (token.getExpires().isBefore(Instant.now())) {
+            throw new EquigyException("OAuth token has expired");
+        }
+        return getByQuery(accountingPointId, queryDate, token.getAccessToken());
+    }
+
+    private URL getByQueryUrl(EAN18 accountingPointId, LocalDate queryDate) throws MalformedURLException {
+        // Querystring parameters don't need encoding, since EAN18 and LocalDate only contain characters 0-9 and '-'
+        return URI.create(baseUri.toString() + "/activatedEnergy/getByQuery"
+                + "?accountingPointId=" + accountingPointId.toString()
+                + "&queryDate=" + queryDate.toString()
+        ).normalize().toURL();
+    }
+
+    public static ActivatedEnergyClient acceptance() {
+        return new ActivatedEnergyClient(URI.create("https://activated-energy-querying-tennet.cbp-acceptance-tennet.ams03.containers.appdomain.cloud/api"));
```

Comment:
Tja, eleganter is het om dit soort strings in n properties file te zetten...

Reply:
Een properties file zou de library wel meteen binden aan specifieke wijze van configureren, bijv. zoals Spring het doet. Omdat de constructor de `public` modifier heeft, is een consumer van de library niet verplicht om deze helper method te gebruiken; er kan simpelweg voor gekozen worden om de constructor rechtstreeks te gebruiken en de endpoint op te nemen in de configuratie.

Reply:
Ja, dat is zo. Een property file in een library is wat minder gangbaar...